providers/system_proxy.rb

```diff
@@ -32,16 +32,17 @@
         }
 
         # Checking params
-        global_settings['http_proxy'] = if not global_settings['http_proxy'].include?('http://')
-          then global_settings['http_proxy'] = 'http://' + global_settings['http_proxy']
-          else global_settings['http_proxy']
+        if !(global_settings['http_proxy'].nil? || global_settings['http_proxy'].include?('http://'))
+           global_settings['http_proxy'] = 'http://'.concat(global_settings['http_proxy'])
         end
 
-        global_settings['https_proxy'] = if not global_settings['https_proxy'].include?('https://')
-          then global_settings['https_proxy'] = 'https://' + global_settings['https_proxy']
-          else global_settings['https_proxy']
+        if !(global_settings['https_proxy'].nil? || global_settings['https_proxy'].include?('https://'))
+           global_settings['https_proxy'] = 'https://'.concat(global_settings['https_proxy'])
         end
-
+        global_settings['http_proxy'] ||= ''
```

Comment:
Default values for global_settings should be assigned before line 35, avoiding the need for testing for nil?. After "http?://" insertion, global_settings should be tested for valid (well formed) URLs and ports. This is a dangerous policy, as it can break connectivity between workstation a Control Centre, and some minimal checks are required.